pkg/config/load.go

```diff
@@ -0,0 +1,203 @@
+package config
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+	"runtime"
+
+	"github.com/cloudposse/atmos/pkg/schema"
+	"github.com/cloudposse/atmos/pkg/ui/theme"
+	u "github.com/cloudposse/atmos/pkg/utils"
+	"github.com/cloudposse/atmos/pkg/version"
+	"github.com/mitchellh/go-homedir"
+	"github.com/spf13/viper"
+)
+
+func LoadConfig(configAndStacksInfo schema.ConfigAndStacksInfo) (schema.AtmosConfiguration, error) {
+	v := viper.New()
+	var atmosConfig schema.AtmosConfiguration
+	v.SetConfigType("yaml")
+	v.SetTypeByDefaultValue(true)
+	setDefaultConfiguration(v)
+	err := readSystemConfig(v)
+	if err != nil {
+		return atmosConfig, err
+	}
+	err = readHomeConfig(v)
+	if err != nil {
+		return atmosConfig, err
+	}
+	err = readWorkDirConfig(v)
+	if err != nil {
+		return atmosConfig, err
+	}
+	err = readEnvAmosConfigPath(v)
+	if err != nil {
+		return atmosConfig, err
+	}
+	if configAndStacksInfo.AtmosCliConfigPath != "" {
+		configFilePath := configAndStacksInfo.AtmosCliConfigPath
+		if len(configFilePath) > 0 {
+			err := mergeConfig(v, configFilePath, CliConfigFileName)
+			switch err.(type) {
+			case viper.ConfigFileNotFoundError:
+				u.LogTrace(fmt.Sprintf("Not Found config file %s", configFilePath))
+			default:
+				return atmosConfig, err
+			}
+		}
+	}
+
+	atmosConfig.CliConfigPath = v.ConfigFileUsed()
+
+	if atmosConfig.CliConfigPath == "" {
+		// If `atmos.yaml` not found, use the default config
+		// Set `ATMOS_LOGS_LEVEL` ENV var to "Debug" to see the message about Atmos using the default CLI config
+		logsLevelEnvVar := os.Getenv("ATMOS_LOGS_LEVEL")
+		if logsLevelEnvVar == u.LogLevelDebug || logsLevelEnvVar == u.LogLevelTrace {
+			u.PrintMessageInColor("'atmos.yaml' CLI config was not found in any of the searched paths: system dir, home dir, current dir, ENV vars.\n"+
+				"Refer to https://atmos.tools/cli/configuration for details on how to configure 'atmos.yaml'.\n"+
+				"Using the default CLI config:\n\n", theme.Colors.Info)
+
+			err = u.PrintAsYAMLToFileDescriptor(atmosConfig, defaultCliConfig)
+			if err != nil {
+				return atmosConfig, err
+			}
+			fmt.Println()
+		}
```

Comment:
I understand you're just copying what we had, but can we move this to native charmbracelet logging